test/blockhistory-migration/BlockHistoryMigrationTests.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE RankNTypes, OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+module Main
+( main
+) where
+
+import Chainweb.Graph
+import Chainweb.Storage.Table.RocksDB
+import Chainweb.Test.TestVersions
+import Chainweb.Test.Utils
+import System.IO.Temp
+import System.LogLevel
+import Test.Tasty
+import Test.Tasty.HUnit
+import qualified Chainweb.Test.MultiNode
+import Chainweb.Version
+import Chainweb.Logger
+import Chainweb.Pact.Backend.Types
+import Chainweb.BlockHeaderDB
+import Control.Monad.IO.Class
+import Chainweb.Test.Pact.Utils
+import System.FilePath
+import Chainweb.Pact.Backend.ChainwebPactDb
+import System.Directory
+import Chainweb.Pact.Backend.Utils
+import Chainweb.PayloadProvider.Pact.BlockHistoryMigration (migrateBlockHistroyTable)
+import Chainweb.Version.EvmTestnet (evmTestnet)
+import Control.Monad.Except
+import qualified Data.ByteString as BS
+import qualified Database.SQLite3.Direct as SQL
+import Chainweb.Version.Mainnet
+import Control.Monad.Trans.Resource
+import qualified Streaming as S
+import Chainweb.Pact.Backend.PactState (qryStream)
+import Data.Function
+import qualified Streaming.Prelude as S
+import Test.HUnit (assertFailure)
+import Chainweb.Utils.Serialization
+import Chainweb.BlockHash
+import Chainweb.TreeDB (lookupRankedM)
+import Chainweb.BlockHeader
+import Control.Lens
+import System.Environment (lookupEnv)
+
+loglevel :: LogLevel
+loglevel = Warn
+
+cid :: ChainId
+cid = unsafeChainId 1
+
+prepareSQLite :: SQLiteEnv -> IO ()
+prepareSQLite sql = either (error "could not prepare sqlitedb") id <$> runExceptT (do
+        exec_ sql
+            "CREATE TABLE IF NOT EXISTS BlockHistory \
+            \(blockheight UNSIGNED BIGINT NOT NULL, \
+            \ endingtxid UNSIGNED BIGINT NOT NULL, \
+            \ hash BLOB NOT NULL, \
+            \ CONSTRAINT blockHeightConstraint UNIQUE (blockheight), \
+            \ CONSTRAINT hashConstraint UNIQUE (hash));"
+        exec_ sql
+            "CREATE TABLE IF NOT EXISTS BlockHistory2 \
+            \(blockheight UNSIGNED BIGINT NOT NULL, \
+            \ endingtxid UNSIGNED BIGINT NOT NULL, \
+            \ hash BLOB NOT NULL, \
+            \ payloadhash BLOB NOT NULL, \
+            \ CONSTRAINT blockHeightConstraint UNIQUE (blockheight), \
+            \ CONSTRAINT hashConstraint UNIQUE (hash));")
```

Comment:
Your note: try to use the existing create table statement for BlockHistory2 to ensure it's accurate.